firedrake/projection.py

```diff
@@ -127,8 +128,16 @@ def __init__(self, v, v_out, solver_parameters=None, constant_jacobian=True):
                              v_out.function_space())
         self.v = v
         self.v_out = v_out
-
-        if not self._same_fspace:
+        if bcs:
+            try:
+                bcs = tuple(bcs)
+            except TypeError:
+                bcs = (bcs,)
+            if any(bc.function_space() != v_out.function_space() for bc in bcs):
+                raise ValueError("bcs must be enforced on the space of the result function.")
+        self.bcs = bcs
+
+        if not self._same_fspace or self.bcs:
```

Comment:
No, I think @thomasgibson is right.  If you have a P1 function space in 1D, and you solve <u, v> = <0, v>, u = 1 on the boundary, you don't get 1, 0, 0, 0, ..., 0, 0, 1, but rather 1, -, +, -, ..., +, -, 1